internal/log/log.go

```diff
@@ -4,65 +4,97 @@ package log
-	Level = logrus.Level
+	Level = slog.Level
 
 	// Handler is the log handler function.
 	Handler = func(_ context.Context, _ Level, format string, args ...interface{})
 )
 
-var (
-	// GetLevel gets the standard logger level.
-	GetLevel = logrus.GetLevel
-	// IsLevelEnabled checks if the log level is greater than the level param.
-	IsLevelEnabled = logrus.IsLevelEnabled
-	// SetFormatter sets the standard logger formatter.
-	SetFormatter = logrus.SetFormatter
-	// SetLevel sets the standard logger level.
-	SetLevel = logrus.SetLevel
-	// SetOutput sets the log output.
-	SetOutput = logrus.SetOutput
-	// SetReportCaller sets whether the standard logger will include the calling method as a field.
-	SetReportCaller = logrus.SetReportCaller
-)
+var logLevelMu = sync.RWMutex{}
+var logLevel slog.Level
+
+var hasCustomOutput atomic.Pointer[io.Writer]
 
 const (
 	// ErrorLevel level. Logs. Used for errors that should definitely be noted.
 	// Commonly used for hooks to send errors to an error tracking service.
-	ErrorLevel = logrus.ErrorLevel
+	ErrorLevel = slog.LevelError
 	// WarnLevel level. Non-critical entries that deserve eyes.
-	WarnLevel = logrus.WarnLevel
+	WarnLevel = slog.LevelWarn
 	// InfoLevel level. General operational entries about what's going on inside the application.
-	InfoLevel = logrus.InfoLevel
+	InfoLevel = slog.LevelInfo
 	// DebugLevel level. Usually only enabled when debugging. Very verbose logging.
-	DebugLevel = logrus.DebugLevel
+	DebugLevel = slog.LevelDebug
 )
 
-func logFuncAdapter(logrusFunc func(args ...interface{})) Handler {
-	return func(_ context.Context, _ Level, format string, args ...interface{}) {
-		logrusFunc(fmt.Sprintf(format, args...))
+func logFuncAdapter(slogFunc func(ctx context.Context, msg string, args ...interface{})) Handler {
+	return func(ctx context.Context, _ Level, format string, args ...interface{}) {
+		slogFunc(ctx, fmt.Sprintf(format, args...))
 	}
 }
 
+var allLevels = []slog.Level{
+	slog.LevelDebug,
+	slog.LevelInfo,
+	slog.LevelWarn,
+	slog.LevelError,
+}
+
 var defaultHandlers = map[Level]Handler{
-	DebugLevel: logFuncAdapter(logrus.Debug),
-	InfoLevel:  logFuncAdapter(logrus.Info),
-	WarnLevel:  logFuncAdapter(logrus.Warn),
-	ErrorLevel: logFuncAdapter(logrus.Error),
+	DebugLevel: logFuncAdapter(slog.DebugContext),
+	InfoLevel:  logFuncAdapter(slog.InfoContext),
+	WarnLevel:  logFuncAdapter(slog.WarnContext),
+	ErrorLevel: logFuncAdapter(slog.ErrorContext),
 }
 var handlers = maps.Clone(defaultHandlers)
 var handlersMu = sync.RWMutex{}
 
+// GetLevel gets the standard logger level.
+func GetLevel() Level {
+	logLevelMu.RLock()
+	defer logLevelMu.RUnlock()
+	return logLevel
+}
+
+// IsLevelEnabled checks if the log level is greater than the level param.
+func IsLevelEnabled(level Level) bool {
+	return isLevelEnabled(context.Background(), level)
+}
+
+func isLevelEnabled(context context.Context, level Level) bool {
+	return slog.Default().Enabled(context, level)
+}
```

Comment:
Why not just exposing this one as I initially did and replace its usage with context?

Reply:
We (currently) never call the exported function with a context. Creating the `context.TODO()` inside this package makes the exported function signature cleaner and simpler to use.

Reply:
Ok, isn't using `context.Background()` though?